CMakeLists.txt

```diff
@@ -1,116 +1,133 @@
+# *                                                                       */
+# * The above copyright notice and this permission notice shall be        */
+# * included in all copies or substantial portions of the Software.       */
+# *                                                                       */
+# * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+# * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+# * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+# * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+# * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+# * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+# * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+# *************************************************************************/
+
+cmake_minimum_required(VERSION 3.0)
+
+include("cmake/HunterGate.cmake")
+HunterGate(
+    URL "https://github.com/ruslo/hunter/archive/v0.20.64.tar.gz"
+    SHA1 "7b830dfda7a094b2af15f44b24ebd2489404d880"
+)
 
-cmake_minimum_required(VERSION 3.1)
-project(spdlog VERSION 0.16.2)
+project(spdlog VERSION 0.16.3)
 include(CTest)
-include(CMakeDependentOption)
-include(GNUInstallDirs)
 
-#---------------------------------------------------------------------------------------
-# compiler config
-#---------------------------------------------------------------------------------------
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+option(SPDLOG_EXTERNAL_TOOLCHAIN "Use an external toolchain, don't set compiler flags" OFF)
 
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-    set(CMAKE_CXX_FLAGS "-Wall ${CMAKE_CXX_FLAGS}")
+if(NOT SPDLOG_EXTERNAL_TOOLCHAIN)
+    set(CMAKE_CXX_STANDARD 11)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+    if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR
+        "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+      set(CMAKE_CXX_FLAGS "-Wall ${CMAKE_CXX_FLAGS}")
+    endif()
 endif()
 
-#---------------------------------------------------------------------------------------
-# spdlog target
-#---------------------------------------------------------------------------------------
 add_library(spdlog INTERFACE)
 
+### fmt library ###
+hunter_add_package(fmt)
+find_package(fmt CONFIG REQUIRED)
+target_link_libraries(spdlog INTERFACE fmt::fmt)
+target_compile_definitions(spdlog INTERFACE SPDLOG_FMT_EXTERNAL=1)
+
 option(SPDLOG_BUILD_EXAMPLES "Build examples" OFF)
-cmake_dependent_option(SPDLOG_BUILD_TESTING
-    "Build spdlog tests" ON
-    "BUILD_TESTING" OFF
-)
+option(SPDLOG_BUILD_TESTS "Build tests" OFF)
 
 target_include_directories(
     spdlog
     INTERFACE
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
-    "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
```

Comment:
This one is fine actually, the value is `include`:
* https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html